protocol/x/clob/memclob/memclob.go

```diff
@@ -1352,18 +1353,23 @@ func (m *MemClobPriceTimePriority) validateNewOrder(
 		return types.ErrInvalidReplacement
 	}
 
-	// If the order is a reduce-only order, we should ensure the full size of the order does not increase the
-	// subaccount's current position size. Note that we intentionally do not validate that the reduce-only order
-	// does not change the subaccount's position _side_, and that will be validated if the order is matched.
-	// TODO(DEC-1228): use `MustValidateReduceOnlyOrder` and move this to `PerformStatefulOrderValidation`.
+	// If the order is a reduce-only order, we should ensure that the sign of the order size is the opposite of
+	// the current position size. Note that we do not validate the size/quantity of the reduce only order fill,
+	// as that will be validated if the order is matched.
+	// The subaccount's current position size is defined as is the current state size + any partial fills
+	// that might have occured as a result of this reduce only order replacing another partially filled order.
```

Comment:
hmm this part doesn't make sense to me. isn't subaccounts position size just stored in state? no need to consider partial fills since they're already written to check state